frameworks/jvm-common/src/main/kotlin/org/apache/amaterasu/frameworks/jvm/common/configuration/dataset/DataSetConfigManager.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.amaterasu.frameworks.jvm.common.configuration.dataset
+
+import java.io.File
+import org.yaml.snakeyaml.Yaml
+import java.io.FileInputStream
+import kotlin.reflect.KClass
+import kotlin.reflect.KParameter
+import kotlin.reflect.full.primaryConstructor
+
+class DataSetConfigManager(env: String, repoPath: String) {
+
+    private val envFolder = "$repoPath/env/$env"
+
+    lateinit var config: Map<String, Config>
+
+    init {
+        config = File(envFolder)
+                .listFiles(DATASET_YAML_FILE_FILTER)
+                .fold(mapOf<String, Config>()) { acc, file -> acc + parseConfigs(file!!) }
+    }
+
+    private inline fun parseConfigs(file: File): Map<String, Config> {
+        FileInputStream(file).use { iStream ->
+            val rawConfigs: Map<String, List<Map<String, String>>> = Yaml().load(iStream)
+            val parsedConfigs = rawConfigs.map { entry ->
+                when (entry.key) {
+                    "file" -> convertMapToConfigType(entry.value, FileConfig::class)
+                    "hive" -> convertMapToConfigType(entry.value, HiveConfig::class)
+                    "generic" -> convertMapToConfigType(entry.value, GenericConfig::class)
```

Comment:
Considering the else, this row is not really needed

Reply:
Not exactly. We said that any non supported configuration has to be available in its raw form in case the user wants to bring his own implementation

Reply:
@nadav-har-tzvi  I think the generic one covers that as well

Reply:
@roadan Yeah I guess.